File: orientation-post-baccalauriat-BackEnd/src/main/java/com/biorbac/model/Institution.java
package com.biorbac.model;

import com.biorbac.enums.InstitutionType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Institution {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long institutionId;
        private String institutionName;
        private String description;
        private String address;
        private String website;
        private Double averageRating;
        private String institutionLogo;


    @Enumerated(EnumType.STRING)
    private InstitutionType institutionType;


    @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL)
    private List<Review> reviews;


    @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL)
    private List<FieldOfStudy> fieldOfStudies;


}

File: orientation-post-baccalauriat-FrontEnd/src/app/core/model/institution.ts
import {FieldOfStudy} from "./field-of-study";
import {Review} from "./review";
import {Student} from "./student";
import {InstitutionType} from "../enums/institution-type";

export interface Institution {
  institutionId:number;
  institutionName: string;
  description: string;
  address: string;
  website: string;
  averageRating:number;
  institutionLogo:string;
  institutionType: InstitutionType;
  student?: Student;
  reviews: Review[];
  fieldOfStudies: FieldOfStudy[];
}
